test/form/widget/hiddenBox/test.js

```diff
@@ -0,0 +1,821 @@
+                        }, 'Init value');
+
+                        return new Promise(function (resolve) {
+                            instance
+                                .off('.test')
+                                .on('change.test', function (value, uri) {
+                                    assert.equal(uri, 'foo', 'The change event has been triggered');
+                                    assert.deepEqual(value, {
+                                        confirmation: '',
+                                        value: ''
+                                    }, 'The expected value is there');
+                                    resolve();
+                                })
+                                .reset();
+                        });
+                    })
+                    .then(function () {
+                        assert.deepEqual(instance.getValue(), {
+                            confirmation: '',
+                            value: ''
+                        }, 'The value has been reset');
+                    })
+                    .catch(function (err) {
+                        assert.ok(false, 'The operation should not fail!');
+                        assert.pushResult({
+                            result: false,
+                            message: err
+                        });
+                    })
+                    .then(function () {
+                        instance
+                            .off('.test')
+                            .destroy();
+                    });
+            })
+            .on('destroy', function () {
+                ready();
+            })
+            .on('error', function (err) {
+                assert.ok(false, 'The operation should not fail!');
+                assert.pushResult({
+                    result: false,
+                    message: err
+                });
+                ready();
+            });
+    });
+
+    QUnit.module('Visual');
+
+    QUnit.test('Visual test', function (assert) {
+        var ready = assert.async();
+        var $container = $('#visual-test .test');
+        var $outputChange = $('#visual-test .change-output');
+        var instance = widgetFactory($container, {
+            widget: 'hidden',
+            uri: 'foo',
+            label: 'Test'
+        });
+
+        assert.expect(3);
+
+        assert.equal($container.children().length, 0, 'The container is empty');
+
+        instance
+            .on('init', function () {
+                assert.equal(this, instance, 'The instance has been initialized');
+            })
+            .on('ready', function () {
+                assert.equal($container.children().length, 2, 'The container contains an element');
+                ready();
+            })
+            .on('change', function (value, uri) {
+                this.validate();
+                $outputChange.val('value of [' + uri + '] changed to "' + value + '"\n' + $outputChange.val());
```

Comment:
Consider replace `value` for `JSON.stringify(value)` once value is an object.

Reply:
Done